pymedphys/_gui/streamlit/pseudonymise.py

```diff
@@ -0,0 +1,297 @@
+
+# hack below copied from content in streamlit forum
+# HACK This only works when we've installed streamlit with pipenv, so the
+# permissions during install are the same as the running process
+STREAMLIT_STATIC_PATH = pathlib.Path(st.__path__[0]) / "static"
+if not STREAMLIT_STATIC_PATH.is_dir():
+    STREAMLIT_STATIC_PATH.mkdir()
+# We create a downloads directory within the streamlit static asset directory
+# and we write output files to it
+DOWNLOADS_PATH = STREAMLIT_STATIC_PATH / "downloads"
+if not DOWNLOADS_PATH.is_dir():
+    DOWNLOADS_PATH.mkdir()
+
+DELAY_TIME = 1
+
+
+def link_to_zipbuffer_download(filename: str, zip_bytes: bytes):
+    """Generates a link allowing the bytes to be downloaded using the specified filename
+    Assumes that the data is in zip format
+
+    Parameters
+    ----------
+    filename : str
+        default filename to use
+    zip_bytes : bytes
+        the bytes from an IOBytes in which a ZipFile wrote its contents, or the bytes
+        read directly from a zip file on disk
+    """
+    b64 = base64.b64encode(zip_bytes).decode()
+    href = f"<a href=\"data:file/zip;base64,{b64}\" download='{filename}'>\
+        Click to download {filename}\
+    </a>"
+    st.sidebar.markdown(href, unsafe_allow_html=True)
+
+
+def link_to_zip_download(zip_to_download: pathlib.Path):
+    """Generates a link allowing the data in a given zip to be downloaded
+
+    Parameters
+    ----------
+    zip_to_download:  pathlib.Path
+        path to zip
+
+    """
+    filename = os.path.basename(zip_to_download)
+    with open(zip_to_download, "rb") as f:
+        zip_bytes = f.read()
+    link_to_zipbuffer_download(filename, zip_bytes)
+
+
+def build_pseudonymised_file_name(ds_input: pydicom.dataset.Dataset):
+    """[summary] Provide full path to the intended destination of the pseudonymised file
+
+    Parameters
+    ----------
+    ds_input : pydicom.dataset.Dataset
+        the non-pseudonymised DICOM data
+
+    Returns
+    -------
+    str
+        full path to the individual pseudonymised file, which can be read later
+        for addition to a zip
+    """
+    pseudo_sop_instance_uid = pseudonymisation_api.pseudonymisation_dispatch["UI"](
+        ds_input.SOPInstanceUID
+    )
+
+    sop_class_uid: pydicom.dataelem.DataElement = ds_input.SOPClassUID
+
+    mode_prefix = DICOM_SOP_CLASS_NAMES_MODE_PREFIXES[
+        sop_class_uid.name  # pylint: disable = no-member
+    ]
+    anon_filename = "{}.{}_Anonymised.dcm".format(mode_prefix, pseudo_sop_instance_uid)
+    temp_anon_filepath = pjoin(DOWNLOADS_PATH, anon_filename)
+    return temp_anon_filepath
+
+
+def _zip_pseudo_fifty_mbytes(
+    file_buffer_list: list, zipfile_path: str, zip_bytes_io=None
+):
+    """Pseudonymises the contents of the file_buffer_list (list of DICOM files)
+    and places the pseudonymised files in to a zip.
+
+    Parameters
+    ----------
+    file_buffer_list: ``list`` of file buffers from streamlit file_uploader
```

Comment:
```suggestion
    file_buffer_list : list
        List of DICOM file buffers from streamlit file_uploader to pseudonymise
```

Portion after ":"  is reserved for type definition. See the parameters section of https://numpydoc.readthedocs.io/en/latest/format.html.